h5p-bildetema/src/components/LanguageSelectorElement/LanguageSelectorElement.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { useL10ns } from "use-h5p";
+import { Language, UserData } from "../../../../common/types/types";
+import { defaultFavouriteLanguages } from "../Bildetema/Bildetema";
+import { languages, languagesOriginal } from "../../constants/languages";
+import { AllowedLanguage } from "../../types/AllowedLanguage";
+import styles from "./LanguageSelectorElement.module.scss";
+
+type LanguageSelectorElement = {
+  language: Language;
+  middleElement: boolean;
+  userData: UserData;
+  setUserData: (updatedUserData: UserData) => void;
+  favLanguages: Language[];
+  setFavLanguages: React.Dispatch<React.SetStateAction<Language[]>>;
+  handleChangeLanguage: (newLanguage: Language) => void;
+};
+
+export const LanguageSelectorElement: React.FC<LanguageSelectorElement> = ({
+  language,
+  middleElement,
+  userData,
+  setUserData,
+  favLanguages,
+  setFavLanguages,
+  handleChangeLanguage,
+}) => {
+  const languageKeys = languages.map(
+    lang => `lang_${lang}`,
+  ) as Array<`lang_${AllowedLanguage}`>;
+
+  const translations = useL10ns(...languageKeys, "selectLanguage");
+
+  const [isChecked, setIsChecked] = React.useState(
+    !!favLanguages.find(favLang => favLang.code === language.code),
+  );
+
+  const handleChange = (): void => {
+    const userDataSnapshot = userData;
+
+    if (
+      userDataSnapshot.favouriteLanguages.find(
+        favLang => favLang.code === language.code,
+      )
+    ) {
```

Comment:
I suggest rewriting for readability, but only if you think it actually makes it more readable.

```suggestion
    const languageIsAlreadyFavorited = userDataSnapshot.favouriteLanguages.find(
        favLang => favLang.code === language.code,
      );
    if (languageIsAlreadyFavorited) {
```